src/middleware/language/language.ts

```diff
@@ -0,0 +1,289 @@
+  } catch {
+    return undefined
+  }
+}
+
+/**
+ * Collection of all language detection strategies
+ */
+export const detectors = {
+  querystring: detectFromQuery,
+  cookie: detectFromCookie,
+  header: detectFromHeader,
+  path: detectFromPath,
+} as const
+
+/** Type for detector functions */
+export type DetectorFunction = (c: Context, options: DetectorOptions) => string | undefined
+
+/** Type-safe detector map */
+export type Detectors = Record<keyof typeof detectors, DetectorFunction>
+
+/**
+ * Validate detector options
+ * @param options Detector options to validate
+ * @throws Error if options are invalid
+ */
+export function validateOptions(options: DetectorOptions): void {
+  if (!options.supportedLanguages.includes(options.fallbackLanguage)) {
+    throw new Error('Fallback language must be included in supported languages')
+  }
+
+  if (options.lookupFromPathIndex < 0) {
+    throw new Error('Path index must be non-negative')
+  }
+
+  if (!options.order.every((detector) => Object.keys(detectors).includes(detector))) {
+    throw new Error('Invalid detector type in order array')
+  }
+}
+
+/**
+ * Cache detected language
+ */
+function cacheLanguage(c: Context, language: string, options: DetectorOptions): void {
+  if (!Array.isArray(options.caches) || !options.caches.includes('cookie')) {
+    return
+  }
+
+  try {
+    setCookie(c, options.lookupCookie, language, options.cookieOptions)
+  } catch (error) {
+    if (options.debug) {
+      console.error('Failed to cache language:', error)
+    }
+  }
+}
+
+/**
+ * Detect language from request
+ */
+const detectLanguage = (c: Context, options: DetectorOptions): string => {
+  let detectedLang: string | undefined
+
+  for (const detectorName of options.order) {
+    const detector = detectors[detectorName]
+    if (!detector) {
+      continue
+    }
+
+    try {
+      detectedLang = detector(c, options)
+      if (detectedLang) {
+        if (options.debug) {
+          console.log(`Language detected from ${detectorName}: ${detectedLang}`)
+        }
+        break
+      }
+    } catch (error) {
+      if (options.debug) {
+        console.error(`Error in ${detectorName} detector:`, error)
+      }
+      continue
+    }
+  }
+
+  const finalLang = detectedLang || options.fallbackLanguage
+
+  if (detectedLang && options.caches) {
+    cacheLanguage(c, finalLang, options)
+  }
+
+  return finalLang
+}
+
+/**
+ * Language detector middleware factory
+ * @param userOptions Configuration options for the language detector
+ * @returns Hono middleware function
+ */
+export const languageDetector = (userOptions: Partial<DetectorOptions>): MiddlewareHandler => {
+  const options: DetectorOptions = {
+    ...DEFAULT_OPTIONS,
+    ...userOptions,
+    cookieOptions: {
+      ...DEFAULT_OPTIONS.cookieOptions,
+      ...userOptions.cookieOptions,
+    },
+  }
+
+  validateOptions(options)
+
+  return async function languageDetectorMiddleware(c, next) {
```

Comment:
Can you rename `languageDetectorMiddleware` to `languageDetector` like other middleware, e.g. [Basic Auth](https://github.com/honojs/hono/blob/main/src/middleware/basic-auth/index.ts#L86)?